### DISCUSSION TITLE
blog/securing-dotnet-webapi-with-amazon-cognito/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#27] [General]
# blog/securing-dotnet-webapi-with-amazon-cognito/

This article is a comprehensive guide on Securing .NET WebAPI with Amazon Cognito. Our focus is on creating a Serverless Authentication system by utilizing OAuth and Amazon Cognito.

https://codewithmukesh.com/blog/securing-dotnet-webapi-with-amazon-cognito/

**Comment by NarenderDhamarlapally** [comment]
Issue with Custom UI Sign-In and Social Identity Login using AWS Cognito
Hi everyone,

I've successfully implemented a custom UI for signing in with username and password using the AWS SDK. However, I'm encountering issues when trying to integrate social identity providers (e.g., Google) for the sign-in process.

Here's what I have so far:

Custom UI for Username and Password: Works perfectly with AWS Cognito using the AWS SDK.
Social Identity Provider (Google): Unable to get it working correctly.
I've set up Google as an identity provider in my Cognito User Pool and configured the necessary OAuth credentials. My front-end successfully redirects to Google's authorization endpoint and returns with an authorization code. The problem arises when I try to exchange this authorization code for tokens and authenticate the user with Cognito.

Request for Assistance
I'm seeking guidance on the following:

Properly handling the Google authorization code exchange and using the resulting tokens with AWS Cognito.
Ensuring the flow works seamlessly within my .NET Core application.
Any potential pitfalls or common issues to look out for in this integration.
Any help or pointers would be greatly appreciated!

Thank you!
